python/pyspark/sql/tests/test_streaming.py

```diff
@@ -592,6 +592,18 @@ def collectBatch(df, id):
             if q:
                 q.stop()
 
+    def test_streaming_foreachBatch_graceful_stop(self):
+        # SPARK-39218: Make foreachBatch streaming query stop gracefully
+        def func(batch_df, _):
+            time.sleep(10)
```

Comment:
How does this func trigger InterruptedException? I would expect codes like `self.spark._jvm.java.lang.Thread.sleep(10000)` instead.

Reply:
Actually, `batch_df.count()` below triggers it via accessing to the interrupted Java thread if I'm not mistaken.

Reply:
I see. `batch_df.count()` probably will touch some code that can be interrupted. Can we use `self.spark._jvm.java.lang.Thread.sleep(10000)` to save 5 seconds in this test?

Reply:
Yeah, that should work too.

Reply:
Let me fix 👍